givewp-example-gateway.php

```diff
@@ -0,0 +1,35 @@
+<?php
+/**
+ * Plugin Name: Give - Example Gateway
+ * Description: Create your own GiveWP payment gateway using this plugin as a guide.
+ * Version: 0.1.0
+ * Requires at least: 5.0
+ * Requires PHP: 7.0
+ * Author: GiveWP - Ben Meredith
+ * Author URI: https://givewp.com
+ * Text Domain: example-give
+ * Domain Path: /languages
+ */
+
+// Register the gateways 
+add_action('givewp_register_payment_gateway', static function ($paymentGatewayRegister) {
+    include 'class-offsite-example-gateway.php';
+    include 'class-onsite-example-gateway.php';
+    $paymentGatewayRegister->registerGateway(ExampleGatewayOffsiteClass::class);
+    $paymentGatewayRegister->registerGateway(ExampleGatewayOnsiteClass::class);
+});
+
+// Filter through the gateway data to add your own data to the $gatewayData param
+add_filter(
+    sprintf("givewp_create_payment_gateway_data_%s", 'onsite-example-test-gateway'),
+    static function ($gatewayData) {
+        // Step 1: Validate the data you need to add to the gatewayData (This comes from your gateway fields).
+        if (isset($_POST['example-gateway-id'])) {
+            // Step 2: Assign the sanitized data to the gatewayData as an array item.
+            $gatewayData['example_payment_id'] = sanitize_text_field($_POST['example-gateway-id']);
+        }
+
+        // Step 3: Return the gatewayData array.
+        return $gatewayData;
+    }
+);
```

Comment:
@jonwaldstein I was hoping we would make this automatic, not requiring this extra hook. The hook was only added for compatibility purposes. I believe if they include a `$_REQUEST['gatewayData']` value it automatically grabs it, right?

Reply:
That's right, they should actually be able to update their input name to be inside gatewayData and they can retrieve it without using the filter.

```
<div><input type='text' name='gatewayData[example-gateway-id]' placeholder='Example gateway field' /></div>
```

Reply:
That worked [b1bc63a](https://github.com/impress-org/givewp-example-gateway/pull/1/commits/b1bc63ab2729f95575a49f15255e3beffc3adebb)